File: justmeet_backend/src/main/java/com/diegodiome/justmeet_backend/config/FirebaseConfig.java
package com.diegodiome.justmeet_backend.config;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;

import javax.annotation.PostConstruct;

import com.google.auth.oauth2.GoogleCredentials;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties
@ConfigurationProperties(prefix = "firebase")
public class FirebaseConfig {

    @PostConstruct
    public void init() {

        try {
            FirebaseApp.getInstance();
        } catch (IllegalStateException e) {
            try {
                FileInputStream inputStream = new FileInputStream(
                        "justmeet-3fd33-firebase-adminsdk-y0dnq-43425c64fd.json");

                try {
                    FirebaseOptions options = new FirebaseOptions.Builder()
                            .setCredentials(GoogleCredentials.fromStream(inputStream)).build();

                    FirebaseApp.initializeApp(options);
                } catch (IOException ioE) {
                    ioE.printStackTrace();
                }
            } catch (NullPointerException nullE) {
                nullE.printStackTrace();
            } catch (FileNotFoundException ex) {
                ex.printStackTrace();
            }
        }

    }
}

File: justmeet_backend/src/main/kotlin/com/diegodiome/justmeet_backend/config/FirebaseConfig.kt
package com.diegodiome.justmeet_backend.config

import com.google.auth.oauth2.GoogleCredentials
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import org.slf4j.LoggerFactory
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.core.io.ClassPathResource
import java.io.IOException
import javax.annotation.PostConstruct

@Configuration
@EnableConfigurationProperties
class FirebaseConfig {

    private var firebaseConfigLogger = LoggerFactory.getLogger(FirebaseConfig::class.java)

    @PostConstruct
    fun init() {
        try {
            val options = FirebaseOptions
                    .builder()
                    .setCredentials(GoogleCredentials
                            .fromStream(ClassPathResource("serviceAccount.json").inputStream))
                    .build()
            FirebaseApp.initializeApp(options)
            firebaseConfigLogger.info("Firebase initialized")
        } catch (ex: IOException) {
            firebaseConfigLogger.warn("Firebase initialization failed : ", ex)
        }
    }
}